bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -280,4 +337,19 @@ private void initializeProperties(Mower mower) {
 
         updateProperties(properties);
     }
+
+    /**
+     * Converts timestamp returned by the Automower API into local time-zone.
+     * Timestamp returned by the API doesn't have offset and it always in the current time zone - it can be treated as
+     * UTC.
+     * Method builds a ZonedDateTime with same hour value but in the current system timezone.
+     * 
+     * @param timestamp - Automower API timestamp
+     * @return ZonedDateTime in system timezone
+     */
+    private ZonedDateTime toZonedDateTime(long timestamp) {
+        ZonedDateTime gmt = ZonedDateTime.ofInstant(Instant.ofEpochMilli(timestamp), ZoneOffset.UTC);
+        return ZonedDateTime.of(gmt.getYear(), gmt.getMonthValue(), gmt.getDayOfMonth(), gmt.getHour(), gmt.getMinute(),
+                gmt.getSecond(), gmt.getNano(), ZoneId.systemDefault());
```

Comment:
The user can set an individual timezone in OH's settings. You can obtain it by injecting `TimeZoneProvider`.
```suggestion
        LocalDateTime local = LocalDateTime.ofInstant(Instant.ofEpochMilli(timestamp), ZoneOffset.UTC);
        return ZonedDateTime.of(local, timeZoneProvider.getTimeZone());
```

Reply:
fixed